### ISSUE TITLE
`npe2==0.2.2` broken with Pyside2 

### ISSUE THREAD

**Issue by Czaki** [napari/npe2#144]
* npe2 version: 0.2.2
* Python version: 3.7-3.9
* Operating System: All

### Description
When running test everything pass for PyQt5 but fail with PySide2 https://github.com/4DNucleome/PartSeg/actions/runs/2085569645
When downgrading `npe2` to version `0.2.1` everything works correctly. 
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/czaki/.pyenv/versions/3.8.3/lib/python3.8/multiprocessing/spawn.py", line 116, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "/home/czaki/.pyenv/versions/3.8.3/lib/python3.8/multiprocessing/spawn.py", line 126, in _main
    self = reduction.pickle.load(from_parent)
  File "/home/czaki/Projekty/PartSeg/package/tests/test_PartSeg/test_sentry.py", line 12, in <module>
    from PartSeg.common_backend.base_argparser import safe_repr
  File "/tmp/tox/py38-PySide2-all/lib/python3.8/site-packages/PartSeg/common_backend/__init__.py", line 14, in <module>
    from napari.settings import get_settings as _napari_get_settings
  File "/tmp/tox/py38-PySide2-all/lib/python3.8/site-packages/napari/settings/__init__.py", line 6, in <module>
    from ._napari_settings import NapariSettings
  File "/tmp/tox/py38-PySide2-all/lib/python3.8/site-packages/napari/settings/_napari_settings.py", line 9, in <module>
    from ._appearance import AppearanceSettings
  File "/tmp/tox/py38-PySide2-all/lib/python3.8/site-packages/napari/settings/_appearance.py", line 4, in <module>
    from ..utils.theme import available_themes
  File "/tmp/tox/py38-PySide2-all/lib/python3.8/site-packages/napari/utils/theme.py", line 335, in <module>
    _install_npe2_themes(_themes)
  File "/tmp/tox/py38-PySide2-all/lib/python3.8/site-packages/napari/utils/theme.py", line 325, in _install_npe2_themes
    import npe2
  File "/tmp/tox/py38-PySide2-all/lib/python3.8/site-packages/shiboken2/files.dir/shibokensupport/__feature__.py", line 142, in _import
    return original_import(name, *args, **kwargs)
  File "/tmp/tox/py38-PySide2-all/lib/python3.8/site-packages/npe2/__init__.py", line 8, in <module>
    from ._dynamic_plugin import DynamicPlugin
  File "/tmp/tox/py38-PySide2-all/lib/python3.8/site-packages/shiboken2/files.dir/shibokensupport/__feature__.py", line 142, in _import
    return original_import(name, *args, **kwargs)
  File "/tmp/tox/py38-PySide2-all/lib/python3.8/site-packages/npe2/_dynamic_plugin.py", line 32, in <module>
    COMMAND_PARAMS = inspect.signature(CommandContribution).parameters
  File "/home/czaki/.pyenv/versions/3.8.3/lib/python3.8/inspect.py", line 3093, in signature
    return Signature.from_callable(obj, follow_wrapped=follow_wrapped)
  File "/home/czaki/.pyenv/versions/3.8.3/lib/python3.8/inspect.py", line 2842, in from_callable
    return _signature_from_callable(obj, sigcls=cls,
  File "/home/czaki/.pyenv/versions/3.8.3/lib/python3.8/inspect.py", line 2252, in _signature_from_callable
    raise TypeError(
TypeError: unexpected object <pydantic.utils.ClassAttribute object at 0x7f4639fda8b0> in __signature__ attribute
```

**Comment by tlambert03** [comment]
Argg, so sorry, thanks for letting me know.  Will try to fix asap, but away at course.  If @Carreau or @DragaDoncila have time to take a look, would be great. Can cut a fix release when ready

**Comment by tlambert03** [comment]
(Btw, this looks like the same issue as in our evented model)

**Comment by Czaki** [comment]
> (Btw, this looks like the same issue as in our evented model)

The last line mentions pydantic, so it may be connected.

**Comment by tlambert03** [comment]
Right, see the context manager in the __new__ method of evented model in napari for details

**Comment by DragaDoncila** [comment]
May have a chance to look at this this afternoon or tomorrow morning if nobody gets to it first.

**Comment by tlambert03** [comment]
fix is in #146
